csharp/src/Apache.Arrow/Arrays/BinaryArray.cs

```diff
@@ -253,14 +253,25 @@ public TBuilder Resize(int length)
 
             public TBuilder Swap(int i, int j)
```

Comment:
I think most users would expect this to swap the ith and jth elements of the binary array and not the ith and jth bytes of the underlying buffer -- or, at least I would. That is, if my logical array were to look like `{null, {0x00, 0x01, 0x02}, {0xff}, {0xfe, 0xfd}}` then calling Swap(1, 2) should result in `{null, {0xff}, {0x00, 0x01, 0x02}, {0xfe, 0xfd}}`. But with this change, calling Swap(1, 2) would instead produce `{null, {0xff}, {0x01, 0x00, 0x02}, {0xfe, 0xfd}}`.